Microsoft.Toolkit.Mvvm/Messaging/WeakReferenceMessenger.cs

```diff
@@ -355,25 +346,92 @@ public bool Remove(TKey key)
             }
 
             /// <inheritdoc cref="IEnumerable{T}.GetEnumerator"/>
-            public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+            [Pure]
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Enumerator GetEnumerator() => new(this);
+
+            /// <summary>
+            /// A custom enumerator that traverses items in a <see cref="ConditionalWeakTable{TKey, TValue}"/> instance.
+            /// </summary>
+            public ref struct Enumerator
```

Comment:
Should we have Unit Tests explicitly for this part?

Reply:
This type is implicitly tested in all the `WeakReferenceMessenger` tests on UWP, as it powers the custom `ConditionalWeakTable<K,V>` implementation we use there, but sure we can add some tests specifically for this if you want (we'll need `[InternalsVisibleTo]` if we don't have it already, as the type is internal). If you're worried about this type being a `ref struct`, that's really only done so that the C# compiler will use pattern matching to get the enumerator when the type is used in a `foreach` statement, so we can skip allocating an object (the `IEnumerator<V>` for the enumerator). It's really just a minor performance improvement we can use here that doesn't affect the overall behavior. I'll go add some tests just in case though, sure! 😄